svm/src/account_loader.rs

```diff
@@ -131,43 +145,123 @@ impl<'a, CB: TransactionProcessingCallback> AccountLoader<'a, CB> {
             .feature_set
             .is_active(&feature_set::disable_account_loader_special_case::id());
 
-        if let Some(account_override) = self
-            .account_overrides
-            .and_then(|overrides| overrides.get(account_key))
-        {
-            Some(LoadedTransactionAccount {
-                loaded_size: account_override.data().len(),
-                account: account_override.clone(),
-                rent_collected: 0,
-            })
-        } else if let Some(program) = (use_program_cache && is_invisible_read)
+        if let Some(program) = (use_program_cache && is_invisible_read)
             .then_some(())
             .and_then(|_| self.program_cache.find(account_key))
         {
             // Optimization to skip loading of accounts which are only used as
             // programs in top-level instructions and not passed as instruction accounts.
-            Some(LoadedTransactionAccount {
+            return Some(LoadedTransactionAccount {
                 loaded_size: program.account_size,
                 account: account_shared_data_from_program(account_key, &self.program_accounts)
                     .ok()?,
                 rent_collected: 0,
-            })
+            });
+        }
+
+        let account = if let Some(account) = self.account_cache.get(account_key) {
+            // Inspect the account prior to collecting rent, since
+            // rent collection can modify the account.
+            self.callbacks
+                .inspect_account(account_key, AccountState::Alive(account), is_writable);
```

Comment:
Shouldn't we check if lamports is 0 before saying the account is alive? Otherwise you could read a dead account into the cache and later write to it and even tho it's still dead, we say it's alive.

Reply:
youre right, ill open a pr for this on monday. thankfully it doesnt affect bank because to get here we would have already had to inspect as writable (which is why this was missed; before removing `AccountCacheItem`, inspecting a previously-alive now-dead account never happened at all)

Reply:
Hmm isn't there a bug post simd-83 where you read a dead account in one tx and then write to it in another? The inspect for the write tx would say it's alive when it's actually dead

Reply:
yes, i made my first edit thinking of that, but made the second edit thinking of the reason why we didnt need to check for equality with `AccountSharedData::default()`. but in this case the account doesnt need to exist

(i have a pr for this now but have been blocked by ci build issues, will add you once its green)